app/models/contact.rb

```diff
@@ -93,26 +80,45 @@ def self.contact_contacts_for(user, person)
     end
   end
 
+  # Follows back if user setting is set so
+  def receive(_recipient_user_ids)
+    user.share_with(person, user.auto_follow_back_aspect) if user.auto_follow_back && !receiving
+  end
+
+  # object for local recipient
+  def object_to_receive
+    Contact.create_or_update_sharing_contact(person.owner, user.person)
+  end
+
+  # @return [Array<Person>] The recipient of the contact
+  def subscribers
+    [person]
+  end
+
+  # creates or updates a contact with active sharing flag. Returns nil if allready sharing.
+  def self.create_or_update_sharing_contact(recipient, sender)
+    contact = recipient.contacts.find_or_initialize_by(person_id: sender.id)
+
+    return if contact.sharing
+
+    contact.tap do |contact|
```

Comment:
You like `.tap`, don't you? :) However, I am not too sure if it adds anything here besides looking fancy. Not using tap and returning the object without tap actually is a line shorter and looks less complicated:

```ruby
contact.sharing = true
contact.save!
contact
```

Also, if you just update a single attribute,

```ruby
contact.update(sharing: true)
contact
```

Would work, too.